moorka-todomvc/src/main/scala/MoorkaTodoMVC.scala

```diff
@@ -89,107 +92,96 @@ object MoorkaTodoMVC extends Application {
-        header(`className` := "header",
-          h1("todos"),
-            block {
-              val inputText = Var("")
-              form(
-                input(
-                  `className` := "new-todo",
-                  `placeholder` := "What needs to be done?",
-                  `autofocus` := true,
-                  `value` =:= inputText
-                ),
-                on.submit listen {
-                  inputText modOnce { inputText ⇒
-                    val s = inputText.trim()
-                    if (s != "") {
-                      todos += Todo(s, Active)
-                    }
-                    ""
+    'div (
+      'section ('className := "todoapp",
+        'header ('className := "header",
+          'h1 ("todos"),
+          'block {
+            val inputText = Var("")
+            'form (
+              'input (
+                'className := "new-todo",
+                'placeholder := "What needs to be done?",
+                'autofocus := true,
+                'value =:= inputText
+              ),
+              'submit listen {
+                inputText modOnce { inputText ⇒
+                  val s = inputText.trim()
+                  if (s != "") {
+                    todos += Todo(s, Active)
                   }
+                  Val("")
                 }
-              )
-            }
-        ),
-        section(`className` := "main",
-          input(
-            `className` := "toggle-all",
-            `type` := "checkbox",
-            `style` := "cursor: pointer",
-            `checked` := {
-              for {
-                todosLength ← todos.rxLength
-                numCompleted ← numCompleted
-              } yield {
-                todosLength > 0 && numCompleted == todosLength
               }
-            },
-            on.click subscribe { event =>
-              `checked` from event.target onSuccess {
-                case true =>
-                  todos.updateAll(_.copy(status = Completed))
-                case false =>
-                  todos.updateAll(_.copy(status = Active))
+            )
+          }
+        ),
+        'section ('className := "main",
+          'input (
+            'className := "toggle-all",
+            'type := "checkbox",
+            'style /= "cursor: pointer",
+            'checked := areAllCompleted,
+            'click listen {
+              areAllCompleted once {
+                case true ⇒ todos.updateAll(_.copy(status = Active))
+                case false ⇒ todos.updateAll(_.copy(status = Completed))
               }
             }
           ),
-          label(`for`:= "toggle-all", "Mark all as complete"),
-          ul(`className` := "todo-list",
+          'label ('htmlFor := "toggle-all", "Mark all as complete"),
+          'ul ('className := "todo-list",
             DataRepeat[Todo](
-              dataProvider = todos,
+              dataProvider = todos, //.filter(_ == filter.unsafeGet),
```

Comment:
Filtered Buffer is not good idea. It is slow. Often I use 'display: none' to hide items from DataRepeat.